autogpt_platform/frontend/src/hooks/useAgentGraph.tsx

```diff
@@ -691,23 +702,23 @@ export default function useAgentGraph(
           const frontendNodeID = blockIDToNodeIDMap[key];
           const frontendNode = prev.find((node) => node.id === frontendNodeID);
 
+          const { position, ...metadata } = backendNode.metadata;
           return frontendNode
-            ? {
+            ? ({
                 ...frontendNode,
-                position: backendNode.metadata.position,
+                position,
                 data: {
                   ...frontendNode.data,
                   hardcodedValues: removeEmptyStringsAndNulls(
                     frontendNode.data.hardcodedValues,
                   ),
                   status: undefined,
                   backend_id: backendNode.id,
-                  webhook: backendNode.webhook,
                   executionResults: [],
-                  metadata: backendNode.metadata,
+                  metadata,
                 },
-              }
-            : null;
+              } satisfies CustomNode)
+            : _backendNodeToXYNode(backendNode, newSavedAgent); // fallback
```

Comment:
this is to ensure a node doesn't just disappear when we can't match backend<->frontend node